model/scope.go

```diff
@@ -0,0 +1,208 @@
+package model
+
+import (
+	"strings"
+)
+
+type Scopes []Scope
```

Comment:
Nit: this feels unnecessary. We can just use `[]Scope` directly.

Reply:
If we do that, Go doesn't allow us to do the beautiful
`func (s []Scope) DoSomething()`
since it is not a named type.

If we were to go that direction, all the functions declared here would need to be written as:
`func DoSomething(s []Scope)`
which would polute a lot the model package and I feel is less elegant.